e2e/complex_service_test.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"context"
+	"testing"
+
+	"github.com/mesg-foundation/engine/hash"
+	pb "github.com/mesg-foundation/engine/protobuf/api"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc/metadata"
+)
+
+func testComplexService(t *testing.T) {
+	var (
+		testServiceHash hash.Hash
+		testRunnerHash  hash.Hash
+	)
+
+	req := newTestComplexCreateServiceRequest()
+
+	t.Run("create", func(t *testing.T) {
+		ctx := metadata.NewOutgoingContext(context.Background(), passmd)
+
+		resp, err := client.ServiceClient.Create(ctx, req)
+		require.NoError(t, err)
+
+		testServiceHash = resp.Hash
+	})
+
+	t.Run("get", func(t *testing.T) {
+		ctx := metadata.NewOutgoingContext(context.Background(), passmd)
+
+		service, err := client.ServiceClient.Get(ctx, &pb.GetServiceRequest{Hash: testServiceHash})
+		require.NoError(t, err)
+		want := pb.TransformCreateReqToService(req)
```

Comment:
I don't like this `protobuf.TransformCreateReqToService`.
The hash calculation should be done by the service sdk.
And I don't think it makes sense to check if the hash is expected. This is a e2e tests, not a unit test.

Reply:
I suggest to revert this function and not check the expected service hash.

Reply:
Ok we don't need to check it. But `TransformCreateReqToService` should there, but maybe without hash calculation. Sdk pachage shoudn't have logic how transform pb structures

Reply:
With the discussion of the last meeting, we spoke about bringing the requests closer to the sdk..

Reply:
The request yes, but this is transformation and it's different.